libs/wingc/src/type_check.rs

```diff
@@ -1522,55 +1480,94 @@ impl<'a> TypeChecker<'a> {
 		}
 	}
 
+	fn expr_maybe_type(&mut self, expr: &Expr, env: &TypeEnv, statement_idx: usize) -> Option<TypeRef> {
+		// TODO: we currently don't handle parenthesized expressions correctly so something like `(MyEnum).A` or `std.(namespace.submodule).A` will return true, is this a problem?
```

Comment:
I would say this is a problem

Reply:
The underlying issue here is that we discard of parentheses info during the parsing phase (because we can just express that in the tree structure). So the end result is that we have the same AST for `(MyEnum).A` and `MyEnum.A`. I don't think it's a big deal to fix. I think opening a separate issue can be enough for now.

Reply:
#1006